### PR TITLE
Fix (transforms): fix bugs in transforms when the gt is empty

### PR DESCRIPTION
**Pull request by ZwwWayne**
Fix https://github.com/open-mmlab/mmdetection/issues/2192, https://github.com/open-mmlab/mmdetection/issues/2171.
This PR fixes the bugs in the transformation when the GT box is empty.
A test function `test_config_data_pipeline` is also added to test the cases when GT is empty and test the correct implementation of the data pipeline.

**Notice**: For now, only augmentation used by SSD and mask RCNN is covered, which are two of the most common augmentation pipelines used in the codebase. For pipelines using `albumentations`, the pipeline will throw None if the GT box is empty. We will try to fix this in a new PR.

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -170,3 +170,210 @@ def test_config_build_detector():
...
+        # 'libra_rcnn/libra_retinanet_r50_fpn_1x.py',
+        # 'libra_rcnn/libra_faster_rcnn_r101_fpn_1x.py',
+        # 'libra_rcnn/libra_faster_rcnn_x101_64x4d_fpn_1x.py',
+        # 'libra_rcnn/libra_fast_rcnn_r50_fpn_1x.py',
+        # ---
+        # 'ghm/retinanet_ghm_r50_fpn_1x.py',
+        # ---
+        # 'fp16/retinanet_r50_fpn_fp16_1x.py',
+        'fp16/mask_rcnn_r50_fpn_fp16_1x.py',
+        # 'fp16/faster_rcnn_r50_fpn_fp16_1x.py'
```
Comment (hellock) [inline tests/test_config.py]: Clean config files.

File: tests/test_config.py [hunk 2]
```diff
@@ -170,3 +170,114 @@ def test_config_build_detector():
...
+    from xdoctest.utils import import_module_from_path
+    from mmdet.datasets.pipelines import Compose
+    import numpy as np
+
+    config_dpath = _get_config_directory()
+    print('Found config_dpath = {!r}'.format(config_dpath))
+
+    # import glob
+    # config_fpaths = list(glob.glob(join(config_dpath, '**', '*.py')))
+    # config_names = [relpath(p, config_dpath) for p in config_fpaths]
```
Comment (hellock) [inline tests/test_config.py]: These comments can be removed.

**Comment by LiangHann** [comment]
After I update the transform.py file to the fixed one, even though there is no "at least one array to stack", but the following error happens:
Traceback (most recent call last):
File "./tools/train.py", line 144, in
main()
File "./tools/train.py", line 140, in main
meta=meta)
File "/home/pichao.wang/mmdetection/mmdet/apis/train.py", line 102, in train_detector
meta=meta)
File "/home/pichao.wang/mmdetection/mmdet/apis/train.py", line 177, in _dist_train
runner.run(data_loaders, cfg.workflow, cfg.total_epochs)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/mmcv/runner/runner.py", line 371, in run
epoch_runner(data_loaders[i], **kwargs)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/mmcv/runner/runner.py", line 275, in train
self.model, data_batch, train_mode=True, **kwargs)
File "/home/pichao.wang/mmdetection/mmdet/apis/train.py", line 75, in batch_processor
losses = model(**data)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/modules/module.py", line 532, in call
result = self.forward(*input, **kwargs)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/parallel/distributed.py", line 447, in forward
output = self.module(*inputs[0], **kwargs[0])
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/modules/module.py", line 532, in call
result = self.forward(*input, **kwargs)
File "/home/pichao.wang/mmdetection/mmdet/core/fp16/decorators.py", line 49, in new_func
return old_func(*args, **kwargs)
File "/home/pichao.wang/mmdetection/mmdet/models/detectors/base.py", line 137, in forward
return self.forward_train(img, img_meta, **kwargs)
File "/home/pichao.wang/mmdetection/mmdet/models/detectors/htc.py", line 296, in forward_train
semantic_feat)
File "/home/pichao.wang/mmdetection/mmdet/models/detectors/htc.py", line 104, in _mask_forward_train
mask_pred = mask_head(mask_feats, last_feat, return_feat=False)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/modules/module.py", line 532, in call
result = self.forward(*input, **kwargs)
File "/home/pichao.wang/mmdetection/mmdet/models/mask_heads/htc_mask_head.py", line 35, in forward
x = self.upsample(x)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/modules/module.py", line 532, in call
result = self.forward(*input, **kwargs)
File "/home/pichao.wang/anaconda3/envs/htc/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 778, in forward
output_padding, self.groups, self.dilation)
RuntimeError: cuDNN error: CUDNN_STATUS_BAD_PARAM

How to deal with this problem?

**Comment by sky-fly97** [comment]
> After I update the transform.py file to the fixed one, even though there is no "at least one array to stack", but the following error happens:

Did you solve this problem? I met it too!!!
